### ISSUE TITLE
cuDNN error

### ISSUE THREAD

**Issue by riktimmondal** [junyanz/BicycleGAN#41]
I am using training dataset of 203 samples and validation of 70 samples.
Is this error due to very less amount of samples?
![screenshot from 2019-01-09 12-50-09](https://user-images.githubusercontent.com/24415468/50883247-8a5d2000-140d-11e9-931c-497275e0f384.png)

**Comment by junyanz** [comment]
This is quite hard to tell. Did you manage to run the code on our dataset (e.g., facades)?

**Comment by riktimmondal** [comment]
Yes its working on your dataset(edges2shoes).I will create some more dataset and check it.

**Comment by junyanz** [comment]
Cool. Maybe you have a broken image or 1 channel image in your dataset. But it can be other reasons.

**Comment by lc82111** [comment]
I have the same problem when training on my own dataset with both input and output channel = 1, and HW 256x256. I use the following bash:
`python ./train.py --gpu_ids -1 --display_id -9 --dataroot /mnt/ssd/IVS_Data_All_20181213 --dataset_mode alphaL --alphaL_data_json_file_full_path /mnt/ssd/IV
S_Data_All_20181213/all_combine_20181229.json --num_threads 0 --name alphaL_bicycle_gan_10_01_2019_15 --model bicycle_gan --netG unet_256 --direction AtoB --c
heckpoints_dir ../checkpoints/alphaL/ --preprocess crop --load_size 480 --crop_size 384 --nz 8 --input_nc 1 --output_nc 1 --niter 100 --niter_decay 100 --save
_epoch_freq 10 --use_dropout`

**Comment by junyanz** [comment]
I will have a look next week. Sorry that I am busy with a paper deadline now.

**Comment by lc82111** [comment]
after some dig, I found the problem may cause by un-full mini-batch.  After I set dataset length to even, (default batch-size=2), It seems the training goes well.

**Comment by junyanz** [comment]
Yes, the code divides a batch into two. I can add an assertion.

**Comment by junyanz** [comment]
I added a [function](https://github.com/junyanz/BicycleGAN/blob/master/train.py#L45) to check if the mini-batch is good for training.

**Comment by lc82111** [comment]
Thanks. I have another question and wait for your reply. Thanks again.
 Dut to fake_B=net_G(A,z) and tanh is the last non-linear-layer,  fake_B should be in range(-1, 1). However the real_B has mean 0.5 and std 0.5, this different value range makes net_D easy to tell their different. Do I understand correctly?

**Comment by junyanz** [comment]
It normalizes an image to [-1, 1]. See this [post](https://discuss.pytorch.org/t/understanding-transform-normalize/21730) for more details.

**Comment by lc82111** [comment]
So, (0.5,0.5,0.5), (0.5,0.5,0.5) is the dataset mean and std respectively?
